lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -247,6 +252,17 @@ getBlockId = lookAhead getBlock *> label "getBlockId" (do
     bytes <- getByteString (fromEnum size)
     return $ Hash $ blake2b256 bytes)
 
+getEpochNo :: HasCallStack => Get EpochNo
+getEpochNo = convert <$> getWord32be
+  where
+    convert epochNo
+        | epochNo > maxEpochNo =
+            error "Epoch number out of bounds when decoding from binary."
+        | otherwise =
+            EpochNo $ fromIntegral epochNo
+    maxEpochNo :: Word32
+    maxEpochNo = fromIntegral @Word31 $ unEpochNo maxBound
```

Comment:
From previous comment, I'd suggest slightly transforming this into:

```hs

mkEpochNo :: HasCallStack => Word32 -> EpochNo
mkEpochNo epochNo 
    | epochNo > maxEpochNo =
        error "Epoch number out of bounds when decoding from binary."
    | otherwise =
        EpochNo $ fromIntegral epochNo
  where
    maxEpochNo :: Word32
    maxEpochNo = fromIntegral @Word31 maxBound
```

And use it consistently everywhere else.

Reply:
Done.